### PR TITLE
[#89] Tests for GitHub commit ingestion 

### PR DESCRIPTION
**Pull request by NoobCrewDelux**
## Description

Uses mock data to test "worker/src/lib/github-commit-tracker.ts". Tests are implemented as given in the task description (issue #89) 

## Solution

Verifies functionality and code integrity iff changes are made to the specific file

### CODE REVIEW

File: apps/worker/src/lib/github-commit-tracker.test.ts [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+import { db, type GitHubRepository } from '@repo/db'
+import * as githubAuthModule from './github-auth'
+import * as githubUtilsModule from './github-utils'
+import { upsertCommit, ingestRepoCommits } from './github-commit-tracker'
+
+vi.mock('./github-auth')
+vi.mock('./github-utils')
+
+describe('github-commit-tracker', () => {
+  const logTest = (name: string) => console.log(`[test] ${name}`)
```
[the remark this answers is not in the document]
Comment (kelvinchen8) [reply apps/worker/src/lib/github-commit-tracker.test.ts]: +1, doesn't vitest already print stuff?

File: apps/worker/src/lib/github-commit-tracker.test.ts [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+    mockUpsertCommit = vi.fn()
+    mockResolveIdentity = vi.fn().mockResolvedValue('identity-1')
+
+    vi.mocked(githubAuthModule.getInstallationOctokit).mockResolvedValue(
+      mockOctokit as unknown as InstallationOctokit
+    )
+    vi.mocked(githubUtilsModule.resolveIdentity).mockImplementation(mockResolveIdentity)
+    vi.mocked(githubUtilsModule.withRateLimit).mockImplementation((fn) => fn())
+    ;(db.commitFact.upsert as unknown as MockFn) = mockUpsertCommit.mockResolvedValue(undefined)
```
Comment (kelvinchen8) [inline apps/worker/src/lib/github-commit-tracker.test.ts]: get changes from the latest main, i added a mock postgres db that we should use for db operations

**Review by kelvinchen8 — CHANGES_REQUESTED** [review]

File: apps/worker/src/lib/github-commit-tracker.test.ts (hunk 2 above)
Comment (kelvinchen8) [reply apps/worker/src/lib/github-commit-tracker.test.ts]: probs shoulda told u before, mb

File: apps/worker/src/lib/github-commit-tracker.integration.test.ts
Comment (kelvinchen8) [inline apps/worker/src/lib/github-commit-tracker.integration.test.ts:1]: maybe add a test that tests if `resolveIdentity` returns null (author cant be resolved)

**Review by kelvinchen8 — CHANGES_REQUESTED** [review]

File: apps/worker/src/lib/github-commit-tracker.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,251 @@
+import { describe, it, expect, beforeEach, vi } from 'vitest'
+import { db } from '@repo/db'
+import { getInstallationOctokit } from './github-auth'
```
Comment (kelvinchen8) [inline apps/worker/src/lib/github-commit-tracker.integration.test.ts]: shouldn't this be `@repo/github` like in `github-PR-tracker.integration.test.ts`? change line 8 too

**Review by kelvinchen8 — CHANGES_REQUESTED** [review]

File: apps/worker/src/lib/github-commit-tracker.integration.test.ts (hunk 3 above)
Comment (NoobCrewDelux) [reply apps/worker/src/lib/github-commit-tracker.integration.test.ts]: oh yea I saw that, I was meant to ask you about it actually but forgot cause I got carried away with the race condition thing

Comment (NoobCrewDelux) [reply apps/worker/src/lib/github-commit-tracker.integration.test.ts]: iirc the reason I had ./github-auth was because repo/github wasnt in main when I started the branch. I just updated my local main branch now and its there.

**Review by kelvinchen8 — APPROVED** [review]
lgtm
